### PR TITLE
RFC: Flag to pass openai creds at invocation

### PR DESCRIPTION
**Pull request by dev2049**
If we like this approach i'll add it to all OpenAI wrappers

### CODE REVIEW

File: tests/unit_tests/llms/test_openai.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import pytest
+
+from langchain import OpenAI
+
+
+@pytest.mark.requires("openai")
+def test_module_var_not_set():
+    import openai
+
+    key = str(uuid.uuid4())
```
Comment (eyurtsev) [inline tests/unit_tests/llms/test_openai.py:12]: nit: let's remove randomness with some arbitrary uuid or fixed string

**Comment by eyurtsev** [comment]
@dev2049 This works, we'll need to figure out how to handle things for serialization.

File: tests/integration_tests/llms/test_openai.py [hunk 2]
```diff
@@ -51,6 +51,13 @@ def test_openai_extra_kwargs() -> None:
         OpenAI(model_kwargs={"model": "text-davinci-003"})
 
 
+def test_openai_call_creds_at_invocation() -> None:
```
Comment (sfc-gh-rmeng) [inline tests/integration_tests/llms/test_openai.py:54]: Thoughts on adding a test to make sure two LLMs can set different credentials, and they don't override each other? 

e.g LLM 1 have a valid credential, LLM 2 doesn't, and LLM 1 can get the string but LLM 2 can't

File: langchain/llms/openai.py [hunk 3]
```diff
@@ -163,6 +163,8 @@ class BaseOpenAI(BaseLLM):
     """Set of special tokens that are allowed。"""
     disallowed_special: Union[Literal["all"], Collection[str]] = "all"
     """Set of special tokens that are not allowed。"""
+    pass_creds_at_invocation: bool = False
```
Comment (sfc-gh-rmeng) [inline langchain/llms/openai.py:166]: maybe add some documentation here?

File: tests/integration_tests/llms/test_openai.py (hunk 2 above)
Comment (dev2049) [reply tests/integration_tests/llms/test_openai.py:54]: i think that's probably already covered by the unit test in file below (if the openai global api key isn't being set then there's no way for one objects instance vars to overwrite another, right?)

Comment (sfc-gh-rmeng) [reply tests/integration_tests/llms/test_openai.py:54]: Yah that works too. I wish there is a more direct integration test, where different OpenAI() llms can be initialized with different api keys and they have isolated behaviors.

**Review by hwchase17 — COMMENTED** [review]
why dont we just always pass at invocation?

**Comment by dev2049** [comment]
> why dont we just always pass at invocation?

openai docs make it seem like setting global vars is proper way to do it but i don't feel strongly about that. makes serialization a bit trickier but that's something we'll have to reckon with anyways.

so mainly just didn't want to rock the boat too much (in case there was a legit reason to not pass at invocation i wasn't aware of)

**Pull request closed without merging**